ui/apps/platform/src/Containers/MainPage/InviteUsers/InviteUsersConfirmationNoEmail.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { ReactElement } from 'react';
+import { Alert, ClipboardCopy, ClipboardCopyVariant, Text } from '@patternfly/react-core';
+import { Link } from 'react-router-dom';
+
+import { accessControlBasePath } from 'routePaths';
+import { BucketsForNewAndExistingEmails } from './InviteUsers.utils';
+
+type InviteUsersConfirmationNoEmailProps = {
+    emailBuckets: BucketsForNewAndExistingEmails;
+    onClose: () => void;
+    role: string;
+};
+
+function InviteUsersConfirmationNoEmail({
+    emailBuckets,
+    onClose,
+    role,
+}: InviteUsersConfirmationNoEmailProps): ReactElement {
+    return (
+        <div>
+            {emailBuckets.newEmails.length === 0 ? (
+                <Alert
+                    title="All entered emails already have auth provider rules"
+                    variant="warning"
+                    isInline
+                    className="pf-u-mb-lg"
+                >
+                    <Text>
+                        You must enter at least one email that does not yet have a rule in the
+                        system.
+                    </Text>
+                    <Text>
+                        Visit the{' '}
+                        <Link onClick={onClose} to={`${accessControlBasePath}/auth-providers`}>
+                            auth provider
+                        </Link>{' '}
+                        section to check which users already have rules.
+                    </Text>
+                </Alert>
+            ) : (
+                <>
+                    {emailBuckets.existingEmails.length > 0 && (
+                        <Alert
+                            title="Some emails already have auth provider rules."
+                            variant="warning"
+                            isInline
+                            className="pf-u-mb-lg"
+                        >
+                            <Text className="pf-u-mb-md">
+                                The following users could not be invited because their emails
+                                already have rules applied to them.
+                            </Text>
+                            <Text className="pf-u-mb-md">
+                                {emailBuckets.existingEmails.join(', ')}
+                            </Text>
```

Comment:
An observation, not a change request for this contribution: flex vertical layout probably provides the margin bottom between items for free.